lib/internal/Magento/Framework/App/Request/Http.php

```diff
@@ -292,13 +286,11 @@ public function initForward()
      * If passed name will be null whole state array will be returned.
      *
      * @param string $name
-     * @return array|string|null
+     * @return mixed|null
      */
     public function getBeforeForwardInfo($name = null)
     {
-        if ($name === null) {
-            return $this->beforeForwardInfo;
-        } elseif (isset($this->beforeForwardInfo[$name])) {
+        if ($name !== null && isset($this->beforeForwardInfo[$name])) {
```

Comment:
As far as I see, you removed returning `$this->beforeForwardInfo`

Reply:
Good catch.  I've updated.